pkg/cli/mapping_server.go

```diff
@@ -117,15 +126,24 @@ func (c *MappingServerCommand) RunUnstarted(ctx context.Context, args []string)
 		return nil, nil, closer, fmt.Errorf("server.NewHandler: %w", err)
 	}
 
-	closer = func() {
-		if err := handler.Cleanup(); err != nil {
-			logger.Errorw("failed to close clean up handler", "error", err)
-		}
-	}
-
 	srv, err := serving.New(c.cfg.Port)
 	if err != nil {
 		return nil, nil, closer, fmt.Errorf("failed to create serving infrastructure: %w", err)
 	}
+
+	closer = func() error {
+		var retErr error
+		if err := successMessenger.Cleanup(); err != nil {
+			retErr = errors.Join(retErr, fmt.Errorf("failed to close success event messenger: %w", err))
+		}
+		if err := failureMessenger.Cleanup(); err != nil {
+			retErr = errors.Join(retErr, fmt.Errorf("failed to close failure event messenger: %w", err))
+		}
+		if err := processor.Stop(); err != nil {
+			retErr = errors.Join(retErr, fmt.Errorf("failed to stop processor: %w", err))
+		}
```

Comment:
Our goal is to remove all this func, and call the client directly to close them here.